### PR TITLE
Move RenderableNode and DetailedNode into render/

### PR DESCRIPTION
**Pull request by tomwilkie**
A continuation of #228

### CODE REVIEW

File: render/detailed_node_test.go [hunk 1]
```diff
@@ -39,14 +43,14 @@ func TestOriginTable(t *testing.T) {
 		serverHostNodeID: {
 			Title:   "Origin Host",
 			Numeric: false,
-			Rows: []report.Row{
-				{"Host name", serverHostName, ""},
+			Rows: []Row{
+				{"Host name", serverHostID, ""},
```
Comment (peterbourgon) [inline render/detailed_node_test.go]: This isn't semantically correct. We currently set host ID == hostname in the probe, but they're not the same concept.

File: render/render_test.go [hunk 2]
```diff
@@ -94,6 +94,7 @@ var (
 					"domain":          "client-54001-domain",
 					"pid":             "10001",
 					report.HostNodeID: clientHostNodeID,
+					"host_name":       clientHostID,
```
Comment (peterbourgon) [inline render/render_test.go]: Similarly, this isn't correct.

File: render/render_test.go [hunk 3]
```diff
@@ -186,11 +209,23 @@ var (
 				},
 			},
 		},
+		Host: report.Topology{
+			Adjacency: report.Adjacency{},
+			NodeMetadatas: report.NodeMetadatas{
+				serverHostNodeID: report.NodeMetadata{
+					"host_name":      serverHostID,
```
Comment (peterbourgon) [inline render/render_test.go]: Similarly, this isn't correct.

File: render/detailed_node_test.go [hunk 4]
```diff
@@ -12,26 +12,30 @@ func TestMakeDetailedNode(t *testing.T) {
...
+			},
 		},
-		clientAddressNodeID: {
+		clientIP: {
 			Title:   "Origin Address",
 			Numeric: false,
-			Rows: []report.Row{
-				{"Host name", clientHostName, ""},
+			Rows: []Row{
+				{"Host name", "client.hostname.com", ""},
```
Comment (peterbourgon) [inline render/detailed_node_test.go]: Why did this get literalized?

**Comment by peterbourgon** [comment]
Gimme another poke when those are resolved...

**Comment by tomwilkie** [comment]
Sorry about that; the origin tests were originally using the report fixture from report/report_fixture_test.go, but I changed them to use the one from render_test.go.

I'll canonicalise a report fixture for the render module in a future changeset.

File: render/detailed_node_test.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package report_test
+package render
```
Comment (peterbourgon) [inline render/detailed_node_test.go]: Please keep this in the _test package. The only reason _test.go files should be in the original package namespace is if they're explicitly testing package/component internals, and even then it's a bad smell unless the package/component is really complex.

File: render/detailed_node_test.go [hunk 6]
```diff
@@ -12,26 +12,30 @@ func TestMakeDetailedNode(t *testing.T) {
...
 
 func TestOriginTable(t *testing.T) {
-	if _, ok := report.OriginTable(reportFixture, "not-found"); ok {
+	if _, ok := OriginTable(rpt, "not-found"); ok {
 		t.Errorf("unknown origin ID gave unexpected success")
 	}
-	for originID, want := range map[string]report.Table{
-		client54001EndpointNodeID: {
+	for originID, want := range map[string]Table{
+		client54001: {
```
Comment (peterbourgon) [inline render/detailed_node_test.go]: client54001 is never an origin node ID, the keys here must take the form xxxNodeID.

File: render/detailed_node_test.go [hunk 7]
```diff
@@ -12,26 +12,30 @@ func TestMakeDetailedNode(t *testing.T) {
...
 			Numeric: false,
-			Rows:    []report.Row{{"Host name", clientHostName, ""}},
+			Rows: []Row{
+				{"Host name", clientHostName, ""},
+				{"PID", "10001", ""},
+				{"Process name", "curl", ""},
+			},
 		},
-		clientAddressNodeID: {
+		clientIP: {
```
Comment (peterbourgon) [inline render/detailed_node_test.go]: Likewise above

File: render/detailed_node_test.go (hunk 5 above)
Comment (tomwilkie) [inline render/detailed_node_test.go]: Right now the tests reference renderTopology, so need to be in the same namespace.

In a future PR I've merged renderTopology into Map.Render, so the tests can be moved out.  I was trying to keep changes to a minimum in this PR.

Comment (peterbourgon) [inline render/detailed_node_test.go]: Can we do that in this PR instead? I'm happy with larger PRs if it means we avoid creating future work or technical debt.

**Comment by peterbourgon** [comment]
Thanks for making it more comprehensive. LGTM!

**Pull request merged**